docs/automated-testing/e2e-testing/recipes/postman-testing.md

```diff
@@ -0,0 +1,230 @@
+2. Locally use Newman to validate tests are working as intended
+
+    ```powershell
+    newman run tests\e2e_Postman_collection.json -e qa.postman_environment.json
+    ```
+
+3. Build a script that automatically executes Postman Test assertions via Newman and Azure CLI.
+    > NOTE: An Azure Service Principal must be setup to continue using azure cli in this CI pipeline example.
+
+    ```powershell (RunPostmanE2eTests.ps1)
+    # Please treat this as pseudocode, and adjust as necessary.
+    ############################################################
+
+    # 1. login to Azure using a Service Principal
+    az login --service-principal -u $APP_ID -p $AZURE_SECRET --tenant $AZURE_TENANT
+    # 2. list app config vars for an environment
+    envVars = az appconfig kv list --name PostmanAppConfig --label $env | ConvertFrom-Json
+    # 3. step through envVars array to get Key Vault uris
+    keyvaultURI = ""
+    @envVars | % {if($_.key -eq 'password'){keyvaultURI = $_.value}} 
+    # 4. parse uris for Key Vault name and secret names
+    # 5. get secret from Key Vault
+    kvsecret = az keyvault secret show --name $secretName --vault-name $keyvaultName --query "value"
+    # 6. set password value to returned Key Vault secret
+    $envVars | % {if($_.key -eq 'password'){$_.value=$kvsecret}}  
+    # 7. create environment file
+    envFile = @{ "_postman_variable_scope" = "environment", "name" = $env, values = @() }
+    foreach($var in $envVars){
+            $envFile.values += @{ key = $var.key; value = $var.value; }
+    }
+    $envFile | ConvertTo-Json -depth 50 | Out-File -encoding ASCII $env.postman_environment.json
+    # 8. install Newman
+    npm install --save-dev newman
+    # 9. run automated E2E tests via Newman
+    node_modules\.bin\newman run tests\e2e_Postman_collection.json -e $env.postman_environment.json
+    ```
+
+4. Create a yaml file and define a step that will run your test script. (ex. A yaml file targeting Azure Devops that runs a Powershell script.)
+
+    ```yaml
+    # Please treat this as pseudocode, and adjust as necessary.
+    ############################################################
+    displayName: 'Run Postman E2E tests'
+    inputs:
+        targetType: 'filePath'
+        filePath: RunPostmanE2eTests.ps1
+    env:
+        APP_ID: $(environment.appId) # credentials for az cli
+        AZURE_SECRET: $(environment.secret)
+        AZURE_TENANT: $(environment.tenant)
+    ```
+
+This approach has the following upside:
+
+- E2E tests can now be run automatically as part of a CI pipeline.
+
+Ending with this approach has the following downside:
+
+- Postman Environment files are no longer being output to a local environment for hands-on manual testing. However, this can be solved by managing 2 scripts. 1 script to output environment variables locally and 1 for ci. Of course, reintroducing the script that outputs environment files invites the opportunity for files with secrets to get checked into git.
```

Comment:
```suggestion
- Postman Environment files are no longer being output to a local environment for hands-on manual testing. However, this can be solved by managing 2 scripts. 1 script to output environment variables locally and 1 for CI. Of course, reintroducing the script that outputs environment files invites the opportunity for files with secrets to get checked into git.
```